### PR TITLE
fix(discord): add Discord.ClientOptions

### PR DESCRIPTION
**Pull request by Chris-GW**
Otherwise, a "TypeError [CLIENT_MISSING_INTENTS]" is thrown due to the upgraded to discord.js v13 in f14a1ca7. For more informations visist [https://discordjs.guide/additional-info/changes-in-v13.html#intents](https://discordjs.guide/additional-info/changes-in-v13.html#intents) 

### Error message without this change
```
error :: ✖ couldn't send discord DM Valid intents must be provided for the Client. {
  "stack": "TypeError [CLIENT_MISSING_INTENTS]: Valid intents must be provided for the Client.
    at Client._validateOptions (/app/node_modules/discord.js/src/client/Client.js:544:13)
    at new Client (/app/node_modules/discord.js/src/client/Client.js:73:10)
    at getDiscordClientAsync (/app/build/src/messaging/discord.js:178:26)
    at sendDMAsync (/app/build/src/messaging/discord.js:86:28)
    at Object.sendDMAndGetResponseAsync (/app/build/src/messaging/discord.js:170:27)
    at Object.getCaptchaInputAsync (/app/build/src/messaging/captcha.js:19:36)
    at Object.handleCaptchaAsync (/app/build/src/store/captcha-handler.js:36:40)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async isItemInStock (/app/build/src/store/lookup.js:338:23)"
}
```

### Testing

run existing test script for notification and captcha

### CODE REVIEW

File: src/messaging/discord.ts [hunk 1]
```diff
@@ -66,13 +69,13 @@ export function sendDiscordMessage(link: Link, store: Store) {
...
           const {id, token} = getIdAndToken(webhook);
-          const client = new Discord.WebhookClient({id, token});
+          const client = new Discord.WebhookClient({id, token}, clientOptions);
 
           promises.push(
             new Promise((resolve, reject) => {
               client
                 .send({
-                  content: notifyText.join(' '),
+                  content: notifyText.length ? notifyText.join(' ') : null,
```
Comment (Chris-GW) [inline src/messaging/discord.ts:78]: this change should also fix the bug reported in #2723: Couldn't send discord message: "Message content must be a non-empty string"

File: src/messaging/discord.ts [hunk 2]
```diff
@@ -7,6 +7,9 @@ import {RawUserData} from 'discord.js/typings/rawDataTypes';
 
 const {notifyGroup, webhooks, notifyGroupSeries} = config.notifications.discord;
 const {pollInterval, responseTimeout, token, userId} = config.captchaHandler;
+const clientOptions: Discord.ClientOptions = {
+  intents: new Discord.Intents(),
```
Comment (Chris-GW) [inline src/messaging/discord.ts:11]: I am not quite certain which Discord gateway intents are here necessary. `Intents.FLAGS.DIRECT_MESSAGES` seems to be a very suitable choice. But when I tested this, it was enough if I didn't specify any intents at all.
Discord.js  documentation also contains the following [warning](https://discordjs.guide/popular-topics/intents.html#enabling-intents):
> Note that discord.js relies heavily on caching to provide its functionality. Some methods that seem unrelated might stop working if certain events do not arrive.

Comment (jef) [reply src/messaging/discord.ts:11]: Yeah, I kind of haphazardly updated to v13 and changed a few types to make it compile. Thanks for fishing through this.

File: src/messaging/discord.ts (hunk 1 above)
Comment (jef) [reply src/messaging/discord.ts:78]: I suppose that makes sense. Weird that there wouldn't be any `content` to begin with, but makes sense.

**Review by jef — APPROVED** [review]
Thank you for the fix! I appreciate it.

**Pull request merged**